code/contracts/ONEWallet2.sol

```diff
@@ -0,0 +1,157 @@
+                } else {
+                    require(sha256(abi.encodePacked(eotp)) == recoveryHash, "Provided secret for recovery is incorrect.");
+                }
+
+                // 5) execute the operation according to its type
+                if(c.operType == OperType.TRANSFER){
+                    _executeTransfer(c.addr, c.amount);                
+                } else if(c.operType == OperType.SET_LAST_RESORT_ADDR){
+                    if(lastResortAddress == address(0x0) && c.addr != address(0x0))
+                        lastResortAddress = c.addr;
+                } else if(c.operType == OperType.RECOVERY){ // we do not care about params in Commits
+                    _drain(); // IH: I do not find any reason for this operation to be executed using OTP.
+                } else {
+                    emit UnknownOperationType(c.operType);                    
+                }                
+                
+                delete commits[otpIdx][i];
+                lastExecutedIdx = otpIdx; // it is not possible to execute any operations using the otpIdx anymore (handled in commit)
+                return true;
+            }
+        }
+        return false;        
+    }
+           
+    /**
+     * Can be called once in a while by the user (relayer) or automatically from reveal().
+     */
+    function cleanUpDailySpendings() public {
+        uint curDay = block.timestamp / 1 days;
+        uint i = cleanupCheckpoint;
+        for (; i <=  cleanupCheckpoint + CLEAN_UP_DAILY_SPENDING_BATCH && i < curDay; i++)            
+            delete dailySpendings[i]; // NOTE: I guess it makes no difference whether it was initialized or not.
+        cleanupCheckpoint = i; // shift clean-up checkpoint
+    }
+
+    function retire() external returns (bool){
+        require(uint32(block.timestamp / interval) - t0 > lifespan, "Too early to retire");        
+        return _drain();
+    }
+    
+    function getCurrentSpending() external view returns (uint){ 
+        return dailySpendings[block.timestamp / 1 days];
+    }    
+
+    //////////////////////// Internal Functions ////////////////////////// 
+
+    function _verifyMerkleProof(bytes32[] calldata neighbors, uint32 index, bytes32 eotp) internal
+    {                
+        require(neighbors.length == height - 1, "Not enough neighbors provided");
+        bytes32 h = sha256(abi.encodePacked(eotp));
+        for (uint8 i = 0; i < height - 1; i++) {
+            if ((index & 0x01) == 0x01) {
+                h = sha256(abi.encodePacked(neighbors[i], h));                
+            } else {
+                h = sha256(abi.encodePacked(h, neighbors[i]));
+            }
+            index >>= 1;
+        }
+        require(root == h, "Proof is incorrect");        
+    }
+
+    function _executeTransfer(address payable dest, uint amount) internal {
+        require(dailySpendings[block.timestamp / 1 days] + amount <= dailyLimit, "Daily limit exceeded.");        
```

Comment:
This and below statement would revert the reveal. The attacker can use the leaked proofs to commit and reveal his altered transaction.

Reply:
You are ritght. Good catch! Will fix it soon.